android/src/androidTest/java/bbct/android/common/navigation/test/NavigateUpFromFilterCardsTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of BBCT for Android.
+ *
+ * Copyright 2018 codeguru <codeguru@users.sourceforge.net>
+ *
+ * BBCT for Android is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BBCT for Android is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package bbct.android.common.navigation.test;
+
+import android.content.Context;
+import android.support.test.espresso.NoMatchingViewException;
+import android.support.test.rule.ActivityTestRule;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import bbct.android.common.R;
+import bbct.android.lite.provider.LiteActivity;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+
+
+public class NavigateUpFromFilterCardsTest {
+    @Rule
+    public ActivityTestRule<LiteActivity> activityActivityTestRule = new ActivityTestRule<LiteActivity>(LiteActivity.class);
+    private Context context = null;
+
+    @Before
+    public void setUp() throws Exception {
+        activityActivityTestRule.getActivity()
+                .getSupportFragmentManager().beginTransaction();
+        context = getInstrumentation().getTargetContext();
+    }
+
+    private void skipSurvey() {
+        try {
+            onView(withText(R.string.survey1)).check(matches(isDisplayed()));
+            onView(withText(R.string.later))
+                    .check(matches(isDisplayed()))
+                    .perform(click());
+        } catch (NoMatchingViewException e) {
+            //view not displayed logic
+        }
+    }
+
+    @Test
+    public void testNavigateUp() {
+        skipSurvey();
+        String expectedTitle = context.getString(R.string.app_name);
+        onView(allOf(withContentDescription(R.string.abc_action_bar_up_description), isDisplayed())).perform(click());
+        onView(allOf(withContentDescription(R.string.filter_cards_title), isDisplayed())).perform(click());
```

Comment:
You don't need to `perform(click())` here. If I understand correctly, this is just waiting for the `FilterCards` page to appear, correct?

Reply:
Still need a small change here.